d3a_interface/sim_results/all_results.py

```diff
@@ -60,6 +62,14 @@ def update(self, area_dict: Dict, core_stats: Dict, current_market_slot: str):
             v.update(area_dict, core_stats, current_market_slot)
         self._update_memory_utilization()
 
+    def update_from_repr(self, area_representation: Dict):
+        """
+        Can be added as an abstract method to the results_abc if needed
+        """
+        for key, value in self.results_mapping.items():
+            if hasattr(value, 'update_from_repr'):
```

Comment:
Rather than doing this, I would create a base class method that does nothing by default, and only override it in subclasses that you actually need it. That way you can avoid using introspection / hasattr.

Reply:
This still needs to be dealt with